source/MaterialXCore/Document.h

```diff
@@ -320,7 +320,10 @@ class Document : public GraphElement
                           const string& node = EMPTY_STRING)
     {
         NodeDefPtr child = addChild<NodeDef>(name);
-        child->setType(type);
+        if (!type.empty() && type != MULTI_OUTPUT_TYPE_STRING)
```

Comment:
Only add an output of the specified type called "out" if a user does not create a type-less NodeDef or a multioutput NodeDef.